### PR TITLE
feat(bug): skip dq entirely for patient with gender U

### PR DESCRIPTION
**Pull request by thomasyopes**
Ref: #1040

### Description

- do not proceed with DQ in CW if patient has gender U

### Tests

- Local
  - [x] patient gender U skips DQ and PD still
- Staging
  - [ ] patient gender U skips DQ and PD still
- Production
  - [ ] patient gender U skips DQ and PD still

### Release Plan

- [ ] Merge this

### CODE REVIEW

**Review by Orta21 — APPROVED** [review]

File: packages/api/src/external/commonwell/document/document-query.ts [hunk 1]
```diff
@@ -135,6 +136,8 @@ export async function queryAndProcessDocuments({
   if (!isCwEnabledForCx) return interrupt(`CW disabled for cx ${cxId}`);
   const isCwQueryEnabled = await isFacilityEnabledToQueryCW(facilityId, patientParam);
   if (!isCwQueryEnabled) return interrupt(`CW disabled for facility ${facilityId}`);
+  const isCwEnabledForPatient = isCommonwellEnabledForPatient(patientParam);
+  if (!isCwEnabledForPatient) return interrupt(`CW disabled for patient, skipping...`);
```
Comment (leite08) [inline packages/api/src/external/commonwell/document/document-query.ts:140]: Maybe we just replace this by a call to `validateCWEnabled()`?

Comment (thomasyopes) [reply packages/api/src/external/commonwell/document/document-query.ts]: Done.

File: packages/api/src/external/commonwell/patient.ts [hunk 2]
```diff
@@ -113,17 +107,17 @@ export async function create({
 }): Promise<{ commonwellPatientId: string; personId: string } | void> {
   const { log, debug } = out(`CW create - M patientId ${patient.id}`);
 
-  const cwCreateEnabled = await validateCWEnabled({
+  const isCwEnabled = await validateCWEnabled({
     patient,
     facilityId,
-    forceCW: forceCWCreate,
+    forceCW: forceCWCreate || Config.isSandbox(),
```
Comment (leite08) [inline packages/api/src/external/commonwell/patient.ts:113]: Why would we check for sandbox, if we don't have CW enabled on that env?

Comment (thomasyopes) [reply packages/api/src/external/commonwell/patient.ts:113]: This is carry-over from the current definition of validateCWEnabled:

```
  const isSandbox = Config.isSandbox();

  if (!isCommonwellEnabledForPatient(patient)) {
    log(`CW disabled for patient, skipping...`);
    return false;
  }

  if (forceCW || isSandbox) {
    log(`CW forced, proceeding...`);
    return true;
  }
```

I don't know why we're returning true when sandbox if it's disabled there...

File: packages/api/src/command/medical/document/document-query.ts [hunk 3]
```diff
@@ -116,7 +115,7 @@ export async function queryDocumentsAcrossHIEs({
 
   const commonwellEnabled = await isCommonwellEnabled();
   if (!cqManagingOrgName) {
-    if (commonwellEnabled || forceCommonwell || Config.isSandbox()) {
```
Comment (thomasyopes) [inline packages/api/src/command/medical/document/document-query.ts:119]: `commonwellEnabled` I believe is true in the sandbox app config

```
  "commonwellFeatureFlag": {
    "enabled": true
  },
```

https://us-west-2.console.aws.amazon.com/systems-manager/appconfig/applications/h3ws9h8/configurationprofiles/7yz6xfu/versions?region=us-west-2&rev1=24

File: packages/api/src/external/commonwell/patient.ts (hunk 2 above)
Comment (thomasyopes) [reply packages/api/src/external/commonwell/patient.ts:113]: Conclusion from slack: leave CW enabled via force for sandbox because we mock CW for sandbox.

Comment (leite08) [reply packages/api/src/external/commonwell/patient.ts:113]: I'd avoid having `forceCWCreate || Config.isSandbox()` in multiple places. Maybe we leave the `|| Config.isSandbox()` that we had on `validateCWEnabled`?

And then add a TSDoc on `validateCWEnabled` explaining why were checking for sandbox.

File: packages/api/src/external/commonwell/patient.ts [hunk 4]
```diff
@@ -284,17 +278,17 @@ export async function update({
 }): Promise<void> {
   const { log, debug } = out(`CW update - M patientId ${patient.id}`);
 
-  const cwUpdateEnabled = await validateCWEnabled({
+  const isCwEnabled = await validateCWEnabled({
     patient,
     facilityId,
-    forceCW: forceCWUpdate,
+    forceCW: forceCWUpdate || Config.isSandbox(),
```
Comment (leite08) [inline packages/api/src/external/commonwell/patient.ts:284]: ditto: keep this in a single place

File: packages/api/src/external/commonwell/patient.ts (hunk 2 above)
Comment (thomasyopes) [reply packages/api/src/external/commonwell/patient.ts]: Done.  I don't really agree with keeping it in the validation function, it's a shared function now and shouldn't have behavior like this IMO, but having to remember to make it true for Sandbox also isn't ideal.

File: packages/api/src/external/commonwell/patient.ts (hunk 4 above)
Comment (thomasyopes) [reply packages/api/src/external/commonwell/patient.ts]: Done.

File: packages/api/src/external/commonwell/patient.ts (hunk 2 above)
Comment (leite08) [reply packages/api/src/external/commonwell/patient.ts]: Yeah, could prob split the function at some point, but I think it's better on the validation than spread around. 👍

**Review by leite08 — APPROVED** [review]

**Pull request merged**